src/backend/termion.rs

```diff
@@ -137,11 +141,20 @@ where
     }
 
     fn get_cursor(&mut self) -> io::Result<(u16, u16)> {
-        termion::cursor::DetectCursorPos::cursor_pos(&mut self.writer).map(|(x, y)| (x - 1, y - 1))
+        match termion::cursor::DetectCursorPos::cursor_pos(&mut self.writer) {
+            Ok((x, y)) => {
+                self.cursor = Some((x - 1, y - 1));
+                Ok((x - 1, y - 1))
+            }
+            Err(_) => self
+                .cursor
+                .ok_or_else(|| io::Error::other("No last known cursor position")),
+        }
```

Comment:
Personally I would keep the map logic and do the match after that. Then there is no duplicate x-1,y-1 logic needed and its only in one place.

Reply:
Kept the match structure as it was, but introduced a local variable that holds the mapped cursor.